### PR TITLE
Separate logger level reload from config reload

### LINKED ISSUE #2453: fleet-server subsystems restart when log level changes
All subsystems such as the ES bulker, API listeners, etc restart when the fleet-server running under the elastic-agent receives a new log level

### PR DESCRIPTION
**Pull request by michel-laterman**
## What is the problem this PR solves?

Subsystems, such as the ES bulker and API listeners restart whenever the log level is changed.

## How does this PR solve the problem?

Split all logger config changes from server config changes so that a change to the fleet-server logging level will not force the api server or other subsystens to restart. Use zerolog.SetGlobalLevel to control log level instead of creating a new global logger when the log level changes.

## How to test this PR locally

Enroll a fleet-server instance that runs under the elastic-agent.
Change the log level through the Kibana UI.
The server restart message:
```
{"log.level":"info","@timestamp":"2023-03-24T20:26:57.639Z","message":"starting server on configuration change","component":{"binary":"fleet-server","dataset":"elastic_agent.fleet_server","id":"fleet-server-default","type":"fleet-server"},"log":{"source":"fleet-server-default"},"service.name":"fleet-server","ecs.version":"1.6.0","ecs.version":"1.6.0"}
```
should not appear as a part of the fleet-server logs.

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- ~~I have made corresponding changes to the documentation~~
- ~~I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #2453

### CODE REVIEW

**Comment by jlind23** [comment]
@michel-laterman can we please use @pazone e2e tests framework to make sure this works as expected? 
Related E2E PR: https://github.com/elastic/elastic-agent/pull/2292

**Comment by michel-laterman** [comment]
@jlind23, 

I don't think we can use it right away, we'll have to add support for running a fleet-server instance under the agent to the framework.
I'll try to add an agent integration test for this in this repo

**Review by michalpristas — APPROVED** [review]
I haven't tested this but change looks good

File: internal/pkg/server/agent_integration_test.go [hunk 1]
```diff
@@ -177,6 +189,7 @@ func TestAgent(t *testing.T) {
 	})
 	require.NoError(t, err)
 	expected = makeExpected(agentID, 1, inputSource, 3, goodSource)
+	expected.Units[0].LogLevel = proto.UnitLogLevel_DEBUG
```
Comment (michel-laterman) [inline internal/pkg/server/agent_integration_test.go:192]: I'm not sure how the v2 mocker works here. When I tried to add create a new `expected` after it went healthy the log level was ignored

**Comment by jlind23** [comment]
FYI after discussing with @michalpristas, he will test this today.

**Comment by michalpristas** [comment]
tested OK 👍 ✅

**Pull request merged**